### ISSUE TITLE
L1 penalty in equation (3)

### ISSUE THREAD

**Issue by ComVisDinh** [mrharicot/monodepth#55]
Dear,

In the file nomodepth_model.py, from line 108 to line 120

```
def get_disparity_smoothness(self, disp, pyramid):
        disp_gradients_x = [self.gradient_x(d) for d in disp]
        disp_gradients_y = [self.gradient_y(d) for d in disp]

        image_gradients_x = [self.gradient_x(img) for img in pyramid]
        image_gradients_y = [self.gradient_y(img) for img in pyramid]

        weights_x = [tf.exp(-tf.reduce_mean(tf.abs(g), 3, keep_dims=True)) for g in image_gradients_x]
        weights_y = [tf.exp(-tf.reduce_mean(tf.abs(g), 3, keep_dims=True)) for g in image_gradients_y]

        smoothness_x = [disp_gradients_x[i] * weights_x[i] for i in range(4)]
        smoothness_y = [disp_gradients_y[i] * weights_y[i] for i in range(4)]
        return smoothness_x + smoothness_y
```

The equation (3) describes the disparity smooth loss is the sum of weighted L1 norm of the disparity gradient. smoothness_x and smoothness_y can contain negative and positive values. Therefore, for my understanding, the above source should change to
     ...   
     return **tf.abs(**smoothness_x**)** + **tf.abs(**smoothness_y**)**

Please correct me if I am wrong.

**Comment by dantkz** [comment]
Both `smoothness_x` and `smoothness_y` are lists. So, `smoothness_x + smoothness_y` just appends the lists together. `tf.abs` is applied later in the code.

**Comment by ComVisDinh** [comment]
I got it. Thank you.
